### PR TITLE
Step 3 : Switch weapons & bonus !

### PR DESCRIPTION
**Pull request by DrKrakus**

### CODE REVIEW

File: oc-game-swift/Game/Game.swift [hunk 1]
```diff
@@ -83,7 +85,7 @@ class Game {
         }
         
         // When all the players are created, go to the hero selection
-        Game.pickAHeroMenu()
+        pickAHeroMenu()
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift:88]: Note : yes you can remove `Game`, write nothing or even `self` as you prefer.

File: oc-game-swift/Game/Game.swift [hunk 2]
```diff
@@ -219,12 +225,58 @@ class Game {
...
+        showEndStats()
+    }
+    
+    /// Random chest can pop
+    private static func isChestPop(_ hero: Hero?) {
+        // Create a random number between 0 and 99
+        let randomNumber = arc4random_uniform(99)
+        
+        // Check if the chest appears
+        guard randomNumber <= 9 else {
```
Comment (jhanzo) [inline oc-game-swift/Game/Game.swift]: this condition isn't very easy to configure, would you 
1/ add a global variable for defining percentage of chest ?
2/ adapt random logic in this condition

File: oc-game-swift/Heroes/Hero.swift [hunk 3]
```diff
@@ -62,19 +62,75 @@ class Hero {
...
         // If the target is dead
         if target.life <= 0 {
             target.isDead = true
         }
     }
+    
+    /// Switching weapon from the random bonus chest
+    func switchWeapon() {
+        // For the healer
+        if let hero = self as? Healer {
```
Comment (jhanzo) [inline oc-game-swift/Heroes/Hero.swift]: maybe you should replace `if let` with a `switch...case...let`

Comment (jhanzo) [reply oc-game-swift/Heroes/Hero.swift]: Moreover, maybe you could think about a shorter method for gathering all your conditions together ? I let you think about it (I have no solution for now)

**Review by jhanzo — CHANGES_REQUESTED** [review]
good stuff ! minor reviews

Next steps, we will look at global architecture, but is is required some times together for discussing about it 😉

File: oc-game-swift/Game/Game.swift (hunk 2 above)
Comment (DrKrakus) [reply oc-game-swift/Game/Game.swift]: Done.

File: oc-game-swift/Heroes/Hero.swift (hunk 3 above)
Comment (DrKrakus) [reply oc-game-swift/Heroes/Hero.swift]: Done

**Pull request merged**